projects/FW_102/src/main.c

```diff
@@ -0,0 +1,57 @@
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void inc_counter_a(SoftTimerId timer_id, void *context);
+static void inc_counter_b(SoftTimerId timer_id, void *context);
+
+int main(void) {
+  // Initialize soft timer, and soft timer dependency (interrupt is needed)
+  interrupt_init();
+  soft_timer_init();
+
+  Counters loop_counter = {
+    .counter_a = 0,
+    .counter_b = 0,
+  };
+
+  // Increment counter a every 0.5s, counter b ever 1s.
+  soft_timer_start_millis(500, inc_counter_a, &loop_counter, 0);
+  soft_timer_start_millis(1000, inc_counter_b, &loop_counter, 0);
```

Comment:
It's good practice to #define constants for arbitrary numbers like these durations instead of just using them. So please add #defines for these durations with a descriptive name that ends with the unit, like `_MS`. Also use them in the `soft_timer_start_millis` call in the functions below.